### DISCUSSION TITLE
CANNOT DEPLOY SMART CONTRACT LOCALLY  (LESSON 5)

### DISCUSSION THREAD

**Discussion by kemsguy7** [Cyfrin/foundry-full-course-cu#1690] [General]
I am getting this error 
```
[⠒] Compiling...
[⠰] Installing Solc version 0.8.25
[⠒] Failed to install Solc 0.8.25: error decoding response body
Error: 
installation failed: error decoding response body
```

When I try deploying my smart contract locally in terminal using 
```
forge script script/DeploySimpleStorage.s.sol --rpc-url http://127.0.0.1:8545 --broadcast --private-key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
```

Please I need help

**Comment by EngrPips** [comment]
Hello @kemsguy7, Please run `forge build` and let me know if that works.

**Reply by kemsguy7** [reply]
No, I'm still seeing the same error

**Reply by EngrPips** [reply]
I would suggest you take a look at this https://ethereum.stackexchange.com/questions/149006/foundry-is-not-able-to-install-solc `ethereum stackexchange` issue and see if it would help you resolve this issue.

**Reply by EngrPips** [reply]
Hello @kemsguy7, You should try out the below and let me know if it works for you.

```
sudo npm uninstall -g solc
```
Then use the below command 
```
npm config delete proxy 
npm config delete https-proxy
```

```
node --trace-deprecation app.js
```

```
npm install -g npm@latest
```

```
export SOLC_BINARY_PATH=/path/to/solc
```

**Comment by tunank** [comment]
OMG I have the same problem too! Did you find the way to solve this ?

**Reply by kemsguy7** [reply]
I've been stuck on it all day , haven't found a solution yet

**Reply by tunank** [reply]
OMG MINE JUST WORKED. Here is what I tried. I uninstalled the solc:
```
sudo npm uninstall -g solc     
```
Then I used these commands:
```
npm config delete proxy
npm config delete https-proxy
```
```
node --trace-deprecation app.js
```
```
npm install -g npm@latest
```
```
export SOLC_BINARY_PATH=/path/to/solc
```
And it just worked. Idk which commands help as I asked chat gpt but I did paste all of them here for you to try. Hopefully you will success 🙏

**Reply by tunank** [reply]
Oh I also tried to restart my laptop and used a different network

**Reply by kemsguy7** [reply]
Thank you @tunank , let me try, I'll give you feedback

**Reply by AbigirlMuchineripi** [reply]
Hie @kemsguy7 , did you fixed the problem?
